PsfLauncher/PsfPowershellScriptRunner.h

```diff
@@ -0,0 +1,305 @@
+#pragma once
+
+
+#include "psf_runtime.h"
+#include "StartProcessHelper.h"
+#include <future>
+#include <memory>
+#include <wil\resource.h>
+
+
+class PsfPowershellScriptRunner
+{
+public:
+	PsfPowershellScriptRunner(const psf::json_object* appConfig, const std::filesystem::path& currentDirectory) :
+		currentDirectory(currentDirectory)
+	{
+		auto startScriptInformationObject = PSFQueryStartScriptInfo();
+		auto endScriptInformationObject = PSFQueryEndScriptInfo();
+
+		//If we want to run at least one script make sure powershell is installed on the computer.
+		if (startScriptInformationObject || endScriptInformationObject)
+		{
+			THROW_HR_IF_MSG(ERROR_NOT_SUPPORTED, !CheckIfPowershellIsInstalled(), "PowerShell is not installed.  Please install PowerShell to run scripts in PSF");
+		}
+
+		auto stopOnScriptErrorObject = appConfig->try_get("stopOnScriptError");
+		if (stopOnScriptErrorObject)
+		{
+			stopOnScriptError = stopOnScriptErrorObject->as_boolean().get();
+		}
+
+		if (startScriptInformationObject)
+		{
+			this->startScriptInformation.reset(startScriptInformationObject);
+		}
+
+		if (endScriptInformationObject)
+		{
+			this->endScriptInformation.reset(endScriptInformationObject);
+		}
+
+
+	}
+
+	//RunStartingScript should return an error only if stopOnScriptError is true
+	HRESULT RunStartingScript()
+	{
+		if (!startScriptInformation)
+		{
+			return ERROR_SUCCESS;
+		}
+
+		const std::wstring& scriptPath = GetScriptPath(*startScriptInformation);
+
+		//This method throws an exception if the script does not exist
+		ThrowIfScriptDoesNotExists(scriptPath);

```

Comment:
>ThrowIfScriptDoesNotExists [](start = 2, length = 26)

In fact, consider making this a function ScriptDoesNotExist which returns a bool and wrap that in a THROW_HR_IF(E_FILE_NOT_FOUND NOT, ScriptDoesNotExist())